inspirehep/submissions/views.py

```diff
@@ -23,20 +23,48 @@
 blueprint = Blueprint("inspirehep_submissions", __name__, url_prefix="/submissions")
 
 
-def send_post_request_to_inspire_next(endpoint, data):
-    headers = {
-        "content-type": "application/json",
-        "Authorization": f"Bearer {current_app.config['AUTHENTICATION_TOKEN']}",
-    }
-    response = requests.post(
-        f"{current_app.config['INSPIRE_NEXT_URL']}{endpoint}",
-        data=json.dumps(data),
-        headers=headers,
-    )
-    return response
+class BaseSubmissionsResource(MethodView):
+    def send_post_request_to_inspire_next(self, endpoint, data):
+        headers = {
+            "content-type": "application/json",
+            "Authorization": f"Bearer {current_app.config['AUTHENTICATION_TOKEN']}",
+        }
+        response = requests.post(
+            f"{current_app.config['INSPIRE_NEXT_URL']}{endpoint}",
+            data=json.dumps(data),
+            headers=headers,
+        )
+        return response
+
+    def get_acquisition_source(self):
+        acquisition_source = dict(
+            email=current_user.email,
+            datetime=datetime.datetime.utcnow().isoformat(),
+            method="submitter",
+            internal_uid=int(current_user.get_id()),
+        )
+
+        orcid = self._get_user_orcid()
+        if orcid:
+            acquisition_source["orcid"] = orcid
+
+        return acquisition_source
+
+    def _get_user_orcid(self):
```

Comment:
we don't need ``_`` here since we moved it

Reply:
it's still bit a of private thing though.

Reply:
it's similar to ``get_acquisition_source``